pkg/util/k8sutil/k8sutil.go

```diff
@@ -264,10 +295,12 @@ func newEtcdPod(m *etcdutil.Member, initialCluster []string, clusterName, state,
 		livenessProbe,
 		readinessProbe)
 
-	volumes := []v1.Volume{
-		{Name: "etcd-data", VolumeSource: v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}},
+	if cs.Pod != nil {
```

Comment:
This changes aren't needed?

Reply:
It is taken care by function `AddEtcdVolumeToPod` in the same file.

Reply:
Sorry. I don't understand your comment?
Why add this change here?

Reply:
Here is 
```
func AddEtcdVolumeToPod(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) {
	vol := v1.Volume{Name: etcdVolumeName}
	if pvc != nil {
		vol.VolumeSource = v1.VolumeSource{
			PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{ClaimName: pvc.Name},
		}
	} else {
		vol.VolumeSource = v1.VolumeSource{EmptyDir: &v1.EmptyDirVolumeSource{}}
	}
	pod.Spec.Volumes = append(pod.Spec.Volumes, vol)
}
```

If it is not PV, it will just append the EmptyDir.

Reply:
I still don't understand it.
Why is this related to PV or emptyDir? This is what I saw on this PR on github:
```
	if cs.Pod != nil {
		container = containerWithRequirements(container, cs.Pod.Resources)
	}
```
It only deals with resources.

Reply:
Ah, i thought you are asking ln 267 & 268.

This line I don't think it has anything to do with this PR scope. It was there mostly because of my cherry pick from the old PR. 

I do think it is a nice thing to have. But not belong to this PR's scope. I can remove it though

Reply:
Please remove this.
It's duplicate. Resource requirements/pod policy are applied outside this func:
https://github.com/coreos/etcd-operator/blob/788063849dac25805e0bcdc637400ba55aebe9a3/pkg/util/k8sutil/k8sutil.go#L338